src/authenticators/direct_authenticator/mod.rs

```diff
@@ -136,22 +141,19 @@ mod test {
         let req_auth = RequestAuth::new(app_name.clone().into_bytes());
         let conn_metadata = None;
 
-        let auth_name = authenticator
+        let application = authenticator
             .authenticate(&req_auth, conn_metadata)
             .expect("Failed to authenticate");
 
-        assert_eq!(auth_name.get_name(), &ApplicationName::from_name(app_name));
-        assert!(!auth_name.is_admin);
+        assert_eq!(application.identity.name, app_name);
+        assert!(!application.is_admin);
 
         let req_auth = RequestAuth::new(admin_name.clone().into_bytes());
-        let auth_name = authenticator
+        let application = authenticator
             .authenticate(&req_auth, conn_metadata)
             .expect("Failed to authenticate");
 
-        assert_eq!(
-            auth_name.get_name(),
-            &ApplicationName::from_name(admin_name)
-        );
-        assert!(auth_name.is_admin);
+        assert_eq!(application.identity.name, admin_name);
```

Comment:
Just for my curiosity: how can this compile if the fields are now private?

Reply:
Interesting, didn't notice that. I think it's because the `DirectAuthenticator` is a descendant of `Authenticator` where `Application` is defined. Has to do with this little snippet in Rust by Example I believe:

> If an item is private, it may be accessed by the current module and its descendants.

Source: https://doc.rust-lang.org/reference/visibility-and-privacy.html

Reply:
Ah interesting!